include/deal.II/lac/block_linear_operator.h

```diff
@@ -425,213 +431,140 @@ lower_triangular_operator(const std::array<std::array<LinearOperator<typename Ra
 /**
  * @relates LinearOperator
  *
- * This function is a specification of the above function that
- * allows to work with block matrices @p block_matrix .
+ * This function implement a back substitution argument to invert an upper
+ * block triangular matrix.
+ * It takes as argement an array of array of LinearOperators @p block_matrix
+ * representing a block matrix and an array of LinearOperators
+ * @p inverse_diagonal representing inverses of digonal blocks of @p block_matrix.
  *
- * @ingroup LAOperators
- */
-
-template <unsigned int n,
-          typename Range = BlockVector<double>,
-          typename Domain = Range,
-          typename BlockMatrix>
-LinearOperator<Range, Domain>
-lower_triangular_operator(const BlockMatrix &);
-
-
-template <unsigned int n, typename Range, typename Domain, typename BlockMatrix>
-LinearOperator<Range, Domain>
-lower_triangular_operator(const BlockMatrix &block_matrix)
-{
-  Assert(block_matrix.n_block_rows() == block_matrix.n_block_cols(),
-         ExcDimensionMismatch(block_matrix.n_block_rows(),block_matrix.n_block_cols()) );
-
-  std::array<std::array<LinearOperator<typename Range::BlockType, typename Domain::BlockType>, n>, n> M;
-  for (unsigned int i = 0; i<n; ++i)
-    {
-      for (unsigned int j = 0; j<i; ++j)
-        M[i][j] = linear_operator<typename Range::BlockType, typename Domain::BlockType>(block_matrix.block(i,j));
-      M[i][i] = null_operator(linear_operator<typename Range::BlockType, typename Domain::BlockType>(block_matrix.block(i,i)).reinit_range_vector);
-    }
-  return lower_triangular_operator<n, Range, Domain>(M);
-}
+ * Let us assume we have a linear system where each coefficient of the system is a
+ * matrix:
 
-/**
- * @relates LinearOperator
+ * A00 x0 + A01 x1 + ... + A0n xn = yn
+ *          A11 x1 + ...          = y1
+ *                          ...     ..
+ *                         Ann xn = yn
  *
- * A function that takes an array of arrays of LinearOperators @p block_matrix
- * and returns its associated upper triangular matrix operator
- * (diagonal is not included).
- *
- * @code
- *  a00 | a01 | a02                 | a01 | a02
- *  ---------------             ---------------
- *  a10 | a11 | a12     ->          |     | a12
- *  ---------------             ---------------
- *  a20 | a21 | a22                 |     |
- * @endcode
+ * First of all, xn = Ann^-1 yn
+ * Then, we can use xn to recover x(n-1):
+ *    x(n-1) = A(n-1)(n-1)^-1 ( y(n-1) - A(n-1)n x(n-1) )
+ * and therefore:
+ *    x0 = A00^-1 ( y0 - A0n xn - ... - A01 x1 )
  *
+ * Caveat: Tvmult and Tvmult_add have not been implemented, yet. This may lead to mistakes.
```

Comment:
Same as above. Here and above I'd also specify what is the behavior of this class when you feed it with an array of operators which represents a full block matrix (i.e., this class ignores all lower blocks of the matrix, and above, this class ignores all upper blocks of the matrix).